cmd/libsnap-confine-private/cgroup-support.h

```diff
@@ -37,4 +37,39 @@ void sc_cgroup_create_and_join(const char *parent, const char *name, pid_t pid);
  **/
 bool sc_cgroup_is_v2(void);
 
+/**
+ * sc_cgroup_mount_snapd_hierarchy mounts /run/snapd/cgroup if one is missing.
+ *
+ * This hierarchy is designed for tracking processes and associating them with
+ * a given executable portion of a snap (either an application or a hook).
+ *
+ * The logic mounts an v1 cgroup hierarchy with name=snapd and without any
+ * controllers. Currently no release agent is set and no release notification
+ * is enabled.
+ *
+ * NOTE: This function mounts v1 cgroup even on systems that otherwise boot in
+ * v2 mode, where systemd is no longer using cgroup v1 at all. The motivation
+ * for this idea is that snapd needs to track processes belonging to specific
+ * snap execution entry point (apps and hooks).
+ *
+ * In v1 mode system uses several cgroups, each with a set of controllers but
+ * leaves out some others (e.g. freezer is never used). In v2 all controllers
+ * live in a unified tree of which there can be only one instance. In
+ * particular, for managing services, systemd is already using this tree and we
+ * cannot interact with it in a meaningful way.
+ *
+ * By mounting a controller-less v1 tree, for as long as the kernel supports
+ * this feature, we can still track processes without affecting or interfering
+ * with systemd.
```

Comment:
thank you for adding this comment explaining the v1 vs v2 situation, this helps a lot